MantidPlot/src/Mantid/MantidUI.cpp

```diff
@@ -459,10 +459,11 @@ void MantidUI::saveNexusWorkspace() { executeSaveNexus(); }
 @param workspaceName :: Name of the workspace to delete
 */
 void MantidUI::deleteWorkspace(const QString &workspaceName) {
-  auto alg = createAlgorithm("DeleteWorkspace");
-  alg->setLogging(false);
-  alg->setPropertyValue("Workspace", workspaceName.toStdString());
-  executeAlgorithmAsync(alg);
+  auto workspaceNameStdString = workspaceName.toStdString();
```

Comment:
I agree with the swap to calling the ADS directly. Using an algorithm to do this seems like overkill.